### PR TITLE
[TTAHUB-949] Fix Goal Delete

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change
- The Goal context menu to Delete or Edit a goal on an AR should now have 'Remove' NOT 'Delete'. If a goal is removed from an AR both the Goals and Objectives created should remain in the DB. But should no longer be associated to the report.
- Fixes issue where creating two goals on an AR and deleting the first. It would delete the last goal and leave the page in a broken state.

## How to test

- Create a goal with both new and existing Goals. Removing these Goals from the report and saving should only remove the association. The Goals and Objectives tables should still have both Goals and Objectives.
- Create multiple goals on the AR and delete the first one. Only the first goal should be deleted and you should be able to delete other goals after.

All other Goal functionality on an AR should work as it did before.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-949

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/GoalForm/ReadOnly.js [hunk 1]
```diff
@@ -1,89 +1,133 @@
-import React, { useRef } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import ContextMenu from '../ContextMenu';
-import Modal from '../Modal';
 import ReadOnlyObjective from './ReadOnlyObjective';
 import './ReadOnly.scss';
 
-export default function ReadOnly({
+export function ReadOnlyGoal({
```
Comment (thewatermethod) [inline frontend/src/components/GoalForm/ReadOnly.js:7]: I would personally prefer this in it's own file if possible

**Review by thewatermethod — APPROVED** [review]

File: frontend/src/components/GoalForm/ReadOnly.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalForm/ReadOnly.js:7]: Anything is possible. :astronaut: :new_moon:

Comment (thewatermethod) [reply frontend/src/components/GoalForm/ReadOnly.js:7]: th-2700848131

Comment (AdamAdHocTeam) [reply frontend/src/components/GoalForm/ReadOnly.js]: I knew you would come around. :alien:

**Pull request merged**